test/version-spec.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+. $(dirname "${BASH_SOURCE[0]}")/common.sh
+
+$DUB add-local "$CURR_DIR/version-spec/newfoo"
+$DUB add-local "$CURR_DIR/version-spec/oldfoo"
+
+[[ $($DUB describe foo | grep path | head -1) == *"/newfoo/"* ]] || false
+[[ $($DUB describe foo@1.0.0 | grep path | head -1) == *"/newfoo/"* ]] || false
+[[ $($DUB describe foo@0.1.0 | grep path | head -1) == *"/oldfoo/"* ]] || false
+
+[[ $($DUB test foo | head -1) == *"/newfoo/" ]] || false
+[[ $($DUB test foo@1.0.0 | head -1) == *"/newfoo/" ]] || false
+[[ $($DUB test foo@0.1.0 | head -1) == *"/oldfoo/" ]] || false
+
+[[ $($DUB lint foo | tail -1) == *"/newfoo/" ]] || false
+[[ $($DUB lint foo@1.0.0 | tail -1) == *"/newfoo/" ]] || false
+[[ $($DUB lint foo@0.1.0 | tail -1) == *"/oldfoo/" ]] || false
+
+[[ $($DUB generate cmake foo | head -1) == *"/newfoo/" ]] || false
+[[ $($DUB generate cmake foo@1.0.0 | head -1) == *"/newfoo/" ]] || false
+[[ $($DUB generate cmake foo@0.1.0 | head -1) == *"/oldfoo/" ]] || false
+
+[[ $($DUB build -n foo | head -1) == *"/newfoo/" ]] || false
+[[ $($DUB build -n foo@1.0.0 | head -1) == *"/newfoo/" ]] || false
+[[ $($DUB build -n foo@0.1.0 | head -1) == *"/oldfoo/" ]] || false
+
+[[ $($DUB run -n foo | tail -1) == 'new-foo' ]] || false
+[[ $($DUB run -n foo@1.0.0 | tail -1) == 'new-foo' ]] || false
+[[ $($DUB run -n foo@0.1.0 | tail -1) == 'old-foo' ]] || false
```

Comment:
What was the intent of those `|| false` ? They don't do anything. `X || false` will be `X` no matter what.

Reply:
I not good in bash, but lines 12, 13, 14, 24, 25, 26 (where used regexp like `*"somestring"`) without `|| false` prints `[ERROR] sh:12 command failed`

full output from my machine:
```
Package is already registered: foo (version: 1.0.0)
Package is already registered: foo (version: 0.1.0)
[ERROR] sh:12 command failed <<<<-- this lines
[ERROR] sh:13 command failed
[ERROR] sh:14 command failed
[ERROR] sh:24 command failed
[ERROR] sh:25 command failed
[ERROR] sh:26 command failed
Deregistered package: foo (version: 1.0.0)
Deregistered package: foo (version: 0.1.0)
Fetching dub 1.9.0...
Fetching dub 1.10.0...
Removing dub in /home/deviator/.dub/packages/dub-1.9.0/dub/
Removed package: 'dub'
Removed dub, version 1.9.0.
Removing dub in /home/deviator/.dub/packages/dub-1.10.0/dub/
Removed package: 'dub'
Removed dub, version 1.10.0.
```
but test passes, in all lines `dub` works correctly... I don't understand why it prints and why it not break test at all.
I think this output wasty and mislead. For other lines `|| false` was added for consistent.